trin-core/src/portalnet/overlay_service.rs

```diff
@@ -3045,6 +3071,25 @@ mod tests {
         }
     }
 
+    #[test]
+    fn test_encode_content_payload() {
+        let content_items: Vec<Bytes> = vec![vec![1, 1].into(), vec![2, 2, 2].into()];
+        let expected_content_payload = hex::decode("02010103020202").unwrap();
+
+        let content_payload = encode_content_payload(content_items).unwrap().to_vec();
+
+        assert_eq!(expected_content_payload, content_payload);
+    }
```

Comment:
It's not obvious to the reader of the test that the method behaves correctly, because the hex-string is not easy to interpret. For encoding/decoding tests, I think it's best to test them together. You start with some value `X`. Then you encode `X`. Then you decode the encoded value, and the output should be `X`. If we had test vectors, then we could use those. If we colocate the encoding and decoding functions, then these tests can reside in the same module.